pages/hub/insights/[insightId]/edit.tsx

```diff
@@ -17,7 +17,7 @@ const EditInsightPage: WithPageLayout = () => {
   const { data: insight, isLoading: insightLoading, isError: insightError } = useInsight(id);
   const { isLoading: insightTeamMembersLoading, isError: insightTeamMembersError } = useInsightMembers(Number(id));
   const insightRepos = insight?.repos.map((repo) => repo.repo_id);
-  const { data: repos } = useRepositories(insightRepos);
+  const { data: repos } = useRepositories(insightRepos, 30, 50);
```

Comment:
Any reason for using `50`? Could we not set it to the amount of repos that are associated with that insight page instead?

For future, we could add a load more... button or paging. cc: @isabensusan

Reply:
We have no clue what the amount is until the data is fetched from the hook. Setting 50 is a temporary fix as I'm sure not many users will have more than that amount of repos for now. We will definitely improve this with a load more paging approach in the future.

Reply:
That's correct. 50 is our upper limit at the moment also regardless of type of user

Reply:
Sounds good!

Reply:
I'm also curious why 50? I think ideally, like Nick said, I would anticipate it loading all. What if i have 51 repos in an insight page? Would the 1 repo that doesn't get included in the list to edit?

We should start thinking about the user flows for people who are going to have alot of repos (for example, there are hundreds of projects in Kubernetes alone and even more in the CNCF landscape)

Reply:
Just seeing the above comments 👍🏼 if the soft limit is 50, let's proceed!